spec/lib/jobs/dell-wsman-inventory-spec.js

```diff
@@ -0,0 +1,269 @@
+        wsmanInventoryJob = new WsmanInventoryJobClass({}, {}, uuid.v4());
+        logger = helper.injector.get('Logger');
+        loggerSpy = sinon.spy(logger.prototype, 'error');
+        doneSpy = sinon.spy(wsmanInventoryJob, '_done');
+        this.sandbox = sinon.sandbox.create();
+    });
+
+    beforeEach(function() {
+        this.sandbox.stub(wsmanBaseJob.prototype, 'checkOBM');
+        this.sandbox.stub(wsmanBaseJob.prototype, 'getIpAddress');
+        this.sandbox.stub(wsmanBaseJob.prototype, '_subscribeHttpResponseUuid');
+        this.sandbox.stub(configuration, 'get');
+        this.sandbox.stub(WsmanInventoryJobClass.prototype, '_handleAsyncRequest');
+        this.sandbox.stub(WsmanInventoryJobClass.prototype, 'handleAsyncResponse');
+        mockWaterline.catalogs = {
+            findLatestCatalogOfSource: this.sandbox.stub(),
+            updateByIdentifier: this.sandbox.stub(),
+            create: this.sandbox.stub()
+        };
+        this.sandbox.stub(wsmanTool.prototype, 'clientRequest');
+    });
+
+    afterEach(function() {
+        this.sandbox.restore();
+    });
+
+    describe('initJob', function() {
+        it('should handle the case of lacking dell configuration.', function() {
+            wsmanBaseJob.prototype.checkOBM.resolves(obm);
+            configuration.get.withArgs('dell').returns(undefined);
+            expect(wsmanInventoryJob._initJob()).to.be.rejectedWith('Dell Inventory web service is not defined in wsmanConfig.json.');
+        });
+
+        it('should handle the case of incorrect obm setting that ip address is undefined.', function() {
+            wsmanBaseJob.prototype.checkOBM.resolves(obm);
+            wsmanBaseJob.prototype.getIpAddress.resolves(undefined);
+            configuration.get.withArgs('dell').returns(dellConfigs);
+            expect(wsmanInventoryJob._initJob()).to.be.rejectedWith('No target IP address.');
+        });
+
+        it('should handle the case of incorrect node type', function() {
+            wsmanInventoryJob.nodeType = 'unknown';
+            wsmanBaseJob.prototype.checkOBM.resolves(obm);
+            wsmanBaseJob.prototype.getIpAddress.resolves('172.31.128.73');
+            configuration.get.withArgs('dell').returns(dellConfigs);
+            expect(wsmanInventoryJob._initJob()).to.be.rejectedWith('Inventory collection for node type (unknown) is not implemented.');
+        });
+
+        it('should init job correctly', function() {
+            wsmanInventoryJob.nodeType = 'compute';
+            wsmanBaseJob.prototype.checkOBM.resolves(obm);
+            wsmanBaseJob.prototype.getIpAddress.resolves('172.31.128.73');
+            configuration.get.withArgs('dell').returns(dellConfigs);
+            return wsmanInventoryJob._initJob().then(function() {
+                expect(wsmanInventoryJob.retrys.length === 4);
+            });
+        });
+    });
+
+    describe('inventoryCallback', function() {
```

Comment:
It's better to remove test case for inventoryCallback function. It doesn't make sense to add test case for it.